Lib/inspect.py

```diff
@@ -1832,9 +1833,11 @@ def _check_class(klass, attr):
             return entry.__dict__[attr]
     return _sentinel
 
+
 @functools.lru_cache()
-def _shadowed_dict_from_mro_tuple(mro):
-    for entry in mro:
+def _shadowed_dict_from_weakref_mro_tuple(*weakref_mro):
+    for weakref_entry in weakref_mro:
+        entry = weakref_entry()
```

Comment:
Normally dereferencing a weakref could return `None`. Here it can't because clearly `klass` is alive in the caller, and will keep its whole MRO alive. Maybe worth a comment?